lib/mzgl/gl/api/GraphicsAPI.h

```diff
@@ -0,0 +1,28 @@
+
+#pragma once
+
+class GraphicsAPI {
+public:
+	GraphicsAPI(Graphics &g)
+		: g(g) {}
+	Graphics &g;
+	virtual ~GraphicsAPI()									 = default;
+	virtual void init()										 = 0;
+	virtual void setBlending(bool shouldBlend)				 = 0;
+	virtual void setLineWidth(float f)						 = 0;
+	virtual void setBlendMode(Graphics::BlendMode blendMode) = 0;
+	virtual void clear(vec4 c)								 = 0;
+
+	virtual void maskOn(const Rectf &r) = 0;
+	virtual void maskOff()				= 0;
+	virtual bool isMaskOn()				= 0;
+	virtual Rectf getMaskRect()			= 0;
+
+	virtual void readScreenPixels(uint8_t *data, const Rectf &r) = 0;
```

Comment:
This function definition seems *very* dangerous. The problems that I can see are:
* data not big enough
* data way too big (how does the user know how much data to pass?)
* no size parameter passed with data to tell the function what the upper limit is
I think I would prefer to see it do something like this:
```
[[nodiscard]] virtual std::vector<uint8_t> readScreenPixels(const Rectf &bounds) =0;
```
or similar

Reply:
ah yes, that would be a nice way to do it - this function isn't used in koala - it was used for screenshotting for debugging stuff - the only thin I would argue it to take out data as a parameter rather than a RVO'd return is that you are likely to be calling this multiple times so may want to reuse the vector - but will c++erize it with a std::vector instead of the raw pointer